subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -143,6 +144,11 @@ static struct k_mutex context_lock;
 #define IS_LISTENING(context) (net_context_get_state(context) == \
 			       NET_CONTEXT_LISTENING)
 
+static inline struct net_context *sock_to_net_ctx(int sock)
+{
+	return z_get_fd_obj(sock, NULL, ENOTSOCK);
```

Comment:
I hope we both understand that passing vtable=NULL is a hack.

Reply:
Yes, we do. I'm already working on a solution for secure sockets, according to [this](https://github.com/zephyrproject-rtos/zephyr/pull/10429#discussion_r228254959) comment.